x-pack/metricbeat/module/azure/client_utils.go

```diff
@@ -222,3 +224,95 @@ func getVM(vmName string, vms []VmResource) (VmResource, bool) {
 	}
 	return VmResource{}, false
 }
+
+// Helper function to generate a string key for the dimensions
+func getDimensionKey(dimensions []Dimension) string {
+	var dimensionKey string
+	for _, dimension := range dimensions {
+		dimensionKey += dimension.Name + ","
+	}
+	return dimensionKey
+}
+
+// Function to get the resource IDs from the batch of metrics
+func getResourceIDs(metrics []Metric) []string {
+	var resourceIDs []string
+	for _, metric := range metrics {
+		resourceIDs = append(resourceIDs, metric.ResourceSubId)
+	}
+	return resourceIDs
+}
+
+// batchMetricIsEmpty will check if the metric value is empty, this seems to be an issue with the azure sdk
+func batchMetricIsEmpty(metric azmetrics.MetricValue) bool {
+	if metric.Average == nil && metric.Total == nil && metric.Minimum == nil && metric.Maximum == nil && metric.Count == nil {
+		return true
+	}
+	return false
+}
+
+func mapBatchMetricValues(client *BatchClient, metricValues azmetrics.MetricData) []MetricValue {
+	var currentMetrics []MetricValue
+	// compare with the previously returned values and filter out any double records
+	for _, v := range metricValues.Values {
+		for _, t := range v.TimeSeries {
+			for _, mv := range t.Data {
+				if batchMetricIsEmpty(mv) {
+					continue
+				}
+				//// remove metric values that are not part of the timeline selected
+				//if mv.TimeStamp.After(startTime) && mv.TimeStamp.Before(endTime) {
+				//	continue
+				//}
+				// define the new metric value and match aggregations values
```

Comment:
Is there something we need to address, or we can safely remove this commented block?

Reply:
removed